packages/astro/config.d.ts

```diff
@@ -46,3 +48,22 @@ export function passthroughImageService(): ImageServiceConfig;
  * Return a valid env field to use in this Astro config for `experimental.env.schema`.
  */
 export const envField: EnvField;
+
+/**
+ * Recursively merges a partial configuration object into a valid default.
+ *
+ * Configuration can be merged both before and after being resolved.
+ */
+export function mergeConfig<C extends AstroConfig | AstroInlineConfig>(
+	defaults: C,
+	overrides: DeepPartial<C>,
```

Comment:
Can we make `C extends AstroConfig | AstroInlineConfig | DeepPartial<AstroConfig | AstroInlineConfig>`? That way we allow `mergeConfig` to be used between two different partial configs too.

Reply:
We don't have any such use case on the code base, but yes, we can.